### PR TITLE
Fix creating documents

### PR DESCRIPTION
**Pull request by open-dynaMIX**

### CODE REVIEW

File: caluma/caluma_form/domain_logic.py [hunk 1]
```diff
@@ -100,29 +101,55 @@ def set_file(validated_data):
...
+        row_document.
+        """
+        answers = []
+        for question in form.questions.filter(
+            Q(default_answer__isnull=False) | Q(type=models.Question.TYPE_FORM)
         ).iterator():
+            if question.type == models.Question.TYPE_FORM:
+                if question.row_form is not None:
+                    document = SaveDocumentLogic._set_default_answers_for_form(
+                        question.row_form, document
```
Comment (anehx) [inline caluma/caluma_form/domain_logic.py:121]: I think you mean `sub_form`

**Review by anehx — CHANGES_REQUESTED** [review]

File: caluma/caluma_form/domain_logic.py (hunk 1 above)
Comment (open-dynaMIX) [reply caluma/caluma_form/domain_logic.py]: Yes, fixed.

File: caluma/caluma_form/domain_logic.py [hunk 2]
```diff
@@ -100,29 +101,55 @@ def set_file(validated_data):
...
                 # In case of table questions, we have to evaluate what default to use.
                 # The default value in a row_document wins over the default_answer of a
                 # row_form question
-                for sub_question in question.row_form.questions.filter(
-                    default_answer__isnull=False
-                ).iterator():
-                    for row in new_answer.documents.iterator():
-                        if not row.answers.filter(question=sub_question).exists():
-                            row.answers.add(
+                for row_doc in new_answer.documents.iterator():
```
[the remark this answers is not in the document]
Comment (open-dynaMIX) [reply caluma/caluma_form/domain_logic.py]: Changed.

**Review by anehx — APPROVED** [review]

**Pull request merged**